workload/aspnetapp.yaml

```
@@ -95,40 +95,33 @@ spec:
     port: 80
     targetPort: 8080
 ---
-apiVersion: networking.k8s.io/v1
-kind: Ingress
+apiVersion: traefik.containo.us/v1alpha1
+kind: IngressRoute
 metadata:
   name: aspnetapp-ingress
   namespace: a0042
   annotations:
-    kubernetes.io/ingress.allow-http: "false"
     # defines controller implementing this ingress resource: https://docs.microsoft.com/en-us/azure/dev-spaces/how-to/ingress-https-traefik
     # ingress.class annotation is being deprecated in Kubernetes 1.18: https://kubernetes.io/docs/concepts/services-networking/ingress/#deprecated-annotation
     # For backwards compatibility, when this annotation is set, precedence is given over the new field ingressClassName under spec.
-    kubernetes.io/ingress.class: traefik-internal
-    traefik.ingress.kubernetes.io/router.entrypoints: websecure
-    traefik.ingress.kubernetes.io/router.tls: "true"
-    traefik.ingress.kubernetes.io/router.tls.options: default
-    traefik.ingress.kubernetes.io/router.middlewares: app-gateway-snet@file, gzip-compress@file
+    kubernetes.io/ingress.class: traefik-internal-crd
 spec:
-  # ingressClassName: "traefik-internal"
-  tls:
-  - hosts:
-      - bu0001a0042.aks-ingress.contoso.com
-        # it is possible to opt for certificate management strategy with dedicated
-        # certificates for each TLS SNI route.
-        # In this Rereference Implementation for the sake of simplicity we use a
-        # wildcard default certificate added at Ingress Controller configuration level which is *.example.com
-        # secretName: <bu0001a0042-00-example-com-tls-secret>
-  rules:
-  - host: bu0001a0042.aks-ingress.contoso.com
-    http:
-      paths:
-      - path: /
-        pathType: Prefix
-        backend:
-          service:
-            name: aspnetapp-service
-            port:
-              number: 80
----
+  entryPoints:
+    - websecure
+  routes:
+  - match: Host(`bu0001a0042.aks-ingress.contoso.com`)
+    kind: Rule
+    services:
+    - name: aspnetapp
+      port: 80
+      namespace: a0042
+      kind: Service
+    middlewares:
+      - name: app-gateway-snet@file
+      - name: gzip-compress@file
+  # it is possible to opt for certificate management strategy with dedicated
+  # certificates for each TLS SNI route.
+  # In this Rereference Implementation for the sake of simplicity we use a
+  # wildcard default certificate added at Ingress Controller configuration level which is *.example.com
```

```suggestion
  # wildcard default certificate added at Ingress Controller configuration level which is *.aks-ingress.contoso.com
```

I know this exists this way in the baseline as well, but maybe we can change it here.